src/wp-includes/class-wp-block.php

```diff
@@ -288,6 +288,19 @@ private function process_block_bindings() {
 			'core/post-date' => array( 'datetime' ),
 		);
 
+		/**
+		 * Filters the supported block attributes for block bindings.
+		 *
+		 * @since 6.9.0
+		 *
+		 * @param array $supported_block_attributes The supported block attributes for block bindings.
+		 *                                          The keys are block names and the values are arrays of attribute names.
+		 */
+		$supported_block_attributes = apply_filters(
+			'block_bindings_supported_block_attributes',
```

Comment:
Alternatively, the filter could be called

```suggestion
			'bindable_block_attributes',
```

Reply:
After looking at the existing implementation, I noticed that there is no need to pass the entire `$supported_block_attributes` structure because the processing happens at the block level. In effect, the filter could operate on the block level `$supported_block_attributes[ $this->name ]`. This would make this filter more general, and make future core refactoring simpler as the value passed would look like:

```php
array( 'url', 'text', 'linkTarget', 'rel' )
```

Reply:
For reference regarding the name for the filter, there is another filter in place here:

https://github.com/WordPress/wordpress-develop/blob/7cf0bb5594ab336a2c919958378035e4bbf46db4/src/wp-includes/class-wp-block-bindings-source.php#L98

Reply:
> After looking at the existing implementation, I noticed that there is no need to pass the entire `$supported_block_attributes` structure because the processing happens at the block level. In effect, the filter could operate on the block level `$supported_block_attributes[ $this->name ]`.

The problem with this is that it's impossible to mark a block attribute as bindable if the block isn't in `$supported_block_attributes` yet, no? I.e. it's easy enough to add another bindable attribute to `core/paragraph` or `core/image` etc, but if we wanted to add an attribute of e.g. `core/pullquote`, how would we do that? Or are you suggesting we iterate over all registered blocks (and not just the ones in `$supported_block_attributes`)?

Reply:
🤦‍♂️ Ohh, it only occurred to me now that we don't have to iterate, since `process_block_bindings` is after all a method of `WP_Block` -- so we're always operating within the context of a block, and thus have its name readily available.

Reply:
https://github.com/WordPress/wordpress-develop/pull/9392/commits/859f579fccffbf16ab610360653ebb3d60f2208f

Reply:
We could consider including the block name in the filter, as it'll be most likely be called on a per-block basis.

Reply:
https://github.com/WordPress/wordpress-develop/pull/9392/commits/943cb064da179760b0bf2f4df1fde3246d02a4be